packages/workbox-webpack-plugin/src/lib/generate-webpack-sw.js

```diff
@@ -0,0 +1,29 @@
+const {readFile} = require('./utils/read-file');
+
+const importScripts = ({
+  workboxSWFilename,
+  manifestFilename,
+}) => `importScripts('${workboxSWFilename}', '${manifestFilename}');`;
+
+const generateSW = ({
```

Comment:
Again, given that workbox-build has a generateSW method and workbox-cli has generate-sw, the naming is confusing and I'm not sure what this function is doing or why it's doing it.

Reply:
@gauntface the problem with using `workbox-build`'s `generateSW` method is that it writes to the file system. The reason I have any of these `generate-template` utils is because I couldn't figure out how to use the existing code without writing to the filesystem.

This `generate-webpack-sw` util is a placeholder intended to highlight one of the issues described in the **blocks** section:

> Utilities often populate then write the populated template to the filesystem. I'm having to duplicate and isolate the populate logic and am referencing the lodash templates across modules.

Maybe I missed it, but does `workbox-cli` have a way to generate the template without writing to fs?

*sidenote: sorry for the single comments. Will answer the other points in a review later*

Reply:
See https://github.com/GoogleChrome/workbox/pull/811/files#r140564059 for what we can do in v3 of`workbox-build`, if that would help.